src/environments/environment.test.ts

```diff
@@ -0,0 +1,19 @@
+import { Profile } from '../app/auth/profiles/profiles.component';
+
+export const AppConfig = {
+  production: false,
```

Comment:
Instead of redefining the entire config, let's import the AppConfig from `environment.ts`, set `test` to true and export the modified AppConfig.

Reply:
Sure 👍

Reply:
@anubh-v 

There seems to be an issue with importing directly from `environment.ts` and performing the deconstructing and override as chrome continuously flags it as a circular dependency and the objects do not get formed properly for use with the application. My guess is that its due to the way that angular does the environment replacements based on the specified configuration.

---

One possible way we could mitigate this is to create an `environment.base.ts` and let that have the properties which `environment.ts` simply uses as its config. Then `environment.tes.ts` and `environment.prod.ts` can take the base config and override with necessary values. But the main downside to that would just be the fact that the `env.base` is a "hacky" solution to allowing us to extend the AppConfig 😅 

---

Another is having an `environment.gen.ts` which contains functions that export an AppConfig depending on the `environment` file. Which could look more like this,

`environment.gen.ts`
```typescript
const BaseConfig = {
  githubUrl: 'https://github.com',
  accessTokenUrl: 'https://catcher-proxy.herokuapp.com/authenticate',
}

export function generateDefaultEnv() {
  return {
    ...BaseConfig,
    production: false,
    test: false,
    clientId: '0cbc5e651d8b01e36687',
    origin: 'http://localhost:4200',
    profiles: [
      <Profile>{
        profileName: 'CATcher',
        encodedText: 'CATcher-org/public_data'
      }
    ]
  }
}

export function generateTestEnv() {
  ...generateDefaultEnv(),
  test: true,
  username: ...
  ...
}
```

`environment.ts`
```typescript
import { generateDefaultEnv } from './environment.gen';

export const AppConfig = {
  ...generateDefaultEnv()
};
```

`environment.test.ts`
```typescript
import { generateTestEnv } from './environment.gen';

export const AppConfig = {
  ...generateTestEnv()
};
```

Seems abit cleaner than the first suggestion.